src/originalBudgetTrackingApp/eventEmitters/EventEmitter.ts

```diff
@@ -19,53 +20,87 @@ export enum EventNames {
   LOG = 'LOG'
 }
 
-interface BudgetTrackingEvent {
-  message?: string;
+export enum AccountType {
+  IMPORTER, EXPORTER
 }
 
-export interface ErrorEvent extends BudgetTrackingEvent {
-  error: Error
+export enum AccountStatus {
+  IDLE, PENDING, IN_PROGRESS, DONE, ERROR
 }
 
-interface ImporterEvent extends BudgetTrackingEvent {
-  id: string
-  name: string
-  companyKey: AccountToScrapeConfig['key']
-}
+export class BudgetTrackingEvent {
+  message: string;
 
-interface ImporterErrorEvent extends ImporterEvent, ErrorEvent {
-}
+  vendorId?: CompanyTypes | OutputVendorName;
+
+  accountStatus?: AccountStatus;
 
-interface ImporterEndEvent extends ImporterEvent {
- transactions: EnrichedTransaction[]
+  error?: Error;
+
+  accountType?: AccountType;
```

Comment:
Ok, it also related to my other comment. I'm not sure how much the Importers and Exporters are similar, and if they should have the same logic.
But lets keep it for now, until we will be smarter.